watertap/examples/flowsheets/nf_dspmde/nf.py

```diff
@@ -0,0 +1,463 @@
+                * mass_flow_in
+                / blk.feed.properties[0].mw_comp[ion]
+            )
+            blk.feed.properties[0].flow_mol_phase_comp["Liq", ion].fix(mol_comp_flow)
+        H2O_mass_frac = 1 - sum(x for x in feed_mass_frac.values())
+        H2O_mol_comp_flow = (
+            H2O_mass_frac
+            * pyunits.kg
+            / pyunits.kg
+            * mass_flow_in
+            / blk.feed.properties[0].mw_comp["H2O"]
+        )
+        blk.feed.properties[0].flow_mol_phase_comp["Liq", "H2O"].fix(H2O_mol_comp_flow)
+
+    if mole_frac is not None:
+        for ion, x in mole_frac.items():
+            blk.feed.properties[0].flow_mol_phase_comp["Liq", ion].fix(x)
+    if conc_mass_phase_comp is not None:
+        for ion, x in conc_mass_phase_comp.items():
+            blk.feed.properties[0].conc_mass_phase_comp["Liq", ion].fix(x)
+            blk.feed.properties[0].flow_mol_phase_comp["Liq", ion].unfix()
+        result = solver.solve(blk.feed)
+        blk.feed.properties[0].conc_mass_phase_comp["Liq", "H2O"].fix()
+        for ion, x in conc_mass_phase_comp.items():
+            blk.feed.properties[0].conc_mass_phase_comp["Liq", ion].unfix()
+            blk.feed.properties[0].flow_mol_phase_comp["Liq", ion].fix()
+            blk.feed.properties[0].flow_mass_phase_comp["Liq", ion].unfix()
+        blk.feed.properties[0].conc_mass_phase_comp["Liq", "H2O"].unfix()
+        blk.feed.properties[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+        blk.feed.properties[0].flow_mol_phase_comp["Liq", "H2O"].fix()
+    blk.feed.properties[0].display()
+    set_NF_feed_scaling(blk)
+    blk.feed.properties[0].assert_electroneutrality(
+        defined_state=True,
+        adjust_by_ion="Cl_-",
+        get_property="flow_mol_phase_comp",
+    )
+
+    blk.feed.properties[0].temperature.fix(298.15)
+    iscale.calculate_scaling_factors(blk)
+    # switching to concetration for ease of adjusting in UI
+    for ion, x in conc_mass_phase_comp.items():
+        blk.feed.properties[0].conc_mass_phase_comp["Liq", ion].unfix()
+        blk.feed.properties[0].flow_mol_phase_comp["Liq", ion].fix()
+
+
+def calc_scale(value):
+    return -1 * math.floor(math.log(value, 10))
+
+
+def set_NF_feed_scaling(blk):
+    _add = 0
+    for index in blk.feed.properties[0].flow_mol_phase_comp:
+        scale = calc_scale(blk.feed.properties[0].flow_mol_phase_comp[index].value)
+        print(f"{index} flow_mol_phase_comp scaling factor = {10**(scale+_add)}")
+        blk.properties.set_default_scaling(
+            "flow_mol_phase_comp", 10 ** (scale + 0), index=index
+        )
+
+
+def add_hardness_constraint(stream):
+    comp_list = stream.properties[0.0].component_list
+    # print("comp_list", comp_list)
+    stream.total_hardness = Var(
+        initialize=0.5,
+        bounds=(0, 10000),
+        domain=NonNegativeReals,
+        units=pyunits.mg / pyunits.kg,
+        doc="Water Hardness as CO3",
+    )
+    stream.total_hardness.unfix()
```

Comment:
Why do you need to unfix? After creating the Var, it should already be unfixed. Is there a special reason for this?